---

title: API tutorial


keywords: fastai
sidebar: home_sidebar

summary: "Turorial on how to interact with the dcl_stats_n_plots API"
description: "Turorial on how to interact with the dcl_stats_n_plots API"
nb_path: "03_api_tutorial.ipynb"
---
<!--

#################################################
### THIS FILE WAS AUTOGENERATED! DO NOT EDIT! ###
#################################################
# file to edit: 03_api_tutorial.ipynb
# command to build the docs after a change: nbdev_build_docs

-->

<div class="container" id="notebook-container">
        
    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

</div>
    {% endraw %}

<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Getting-started">Getting started<a class="anchor-link" href="#Getting-started"> </a></h2><p>dcl_stats_n_plots is available from PyPi, so just run:</p>
<blockquote><p><code>pip install dcl-stats-n-plots</code></p>
</blockquote>
<p>for installation. If you are new to python and/or the jupyter interface and want a more detailed guide, please have a look at our <a href="https://dsegebarth.github.io/dcl_stats_n_plots/gui_tutorial.html#Getting-started">installation guide</a>. For this tutorial, we will use some demo files with dummy data, which are available <a href="https://github.com/DSegebarth/dcl_stats_n_plots/tree/master/test_data">here</a>.
In the following, you will find some examples for each type of implemented statistical analysis, including the corresponding plotting and annotation functions. Currently, the implemented analyses are:</p>
<ol>
<li>Comparison of two or more independent samples</li>
<li>Comparison of one group to a reference value</li>
<li>Mixed-Model ANOVA (comparison of two or more groups with repeated measures)</li>
</ol>
<p>You will find the most detailed description of how to use our API in the <a href="https://dsegebarth.github.io/dcl_stats_n_plots/api_tutorial.html#Comparison-of-two-or-more-independent-samples">first section</a>. Thus, you might want to go through this section to get the most in-depth intro to our API, even though you want to perform a different kind of analysis in the end. 
{% include note.html content='This package relies heavily on the correct design of the input data. Therefore, please take some time before using one of the implemented functionalities and make sure you got the input data organized in the right way. In general, this packages requires the input data to be organized in the so-called "long format" with a fixed column order. Please check out the "Expected input data format" section for the respective test for detailed information.' %}</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Comparison-of-two-or-more-independent-samples">Comparison of two or more independent samples<a class="anchor-link" href="#Comparison-of-two-or-more-independent-samples"> </a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Representative-experiment">Representative experiment<a class="anchor-link" href="#Representative-experiment"> </a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>The comparison of two or more independent samples is probably one of the most frequently performed analyses in the lab. You can use this type of analysis if you:</p>
<ul>
<li>have multiple experimental subjects (&gt;3) </li>
<li>that belong to at least two different groups</li>
<li>and the given experiment is perfomed only once with each experimental subject (no repeated measurements)</li>
</ul>
<p>For instance, if you recorded a single open field session of each mouse in a experimental cohort of 36 mice, which represent three different genotypes (12 homozygous transgenic mice [tg/tg], 12 heterozygous transgenic mice [tg/+], and 12 wildtype mice [+/+]). After you analyzed the videos, you now want to test for significant differences in the time the mice spent in the center of the open field - based on their genotype.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>{% include note.html content='Equal samples sizes per group is ideal, yet no prerequisite for this analysis.' %}</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Expected-input-data-format">Expected input data format<a class="anchor-link" href="#Expected-input-data-format"> </a></h3><p>For this analysis, you only have to provide two columns. The first one that contains the actual data values, and the second column that specifies to which group the respective data value belongs. In Microsoft Excel, this might look like the following:</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p><img src="https://raw.githubusercontent.com/DSegebarth/dcl_stats_n_plots/master/media/independent_samples_input_schema.png" alt="independent_samples_input_schema.png"></p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>Where the first two data values belong to the group called "group_a", the third and fourth to a second group called "group_b", and the sixth and seventh value belong to a third group called "group_c". You probably got the idea ;-)</p>
<p>You are entirely free to choose any column names (for instance "time spent in center [s]" instead of "data", or "genotype" instead of "group_id") and also to provide your favorite group names (e.g. "+/+" instead of "group_a" and "tg/+" for "group_b"). 
{% include note.html content='The column name of your data column will be used as the default y-axis label, but you will still be able to modify that on demand.' %}{% include note.html content='The order of appearance of the groups in the group column (top to bottom) will be used as the default x-axis order, but you will still be able to modify that on demand.' %}</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Compute-the-stats">Compute the stats<a class="anchor-link" href="#Compute-the-stats"> </a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>Now that we know how the input data has to be organized, letÂ´s start with the statistical analysis of our dataset, using one of our dummy data test files, which are available <a href="https://github.com/DSegebarth/dcl_stats_n_plots/tree/master/test_data">here</a>. This dataset contains a total of 36 data values from three different groups ('tg/tg', 'tg/+', '+/+' - 12 values per group). For reasons of clarity, only the first 5 rows will be displayed.</p>

</div>
</div>
</div>
    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>

<span class="n">file_url</span> <span class="o">=</span> <span class="s1">&#39;https://raw.githubusercontent.com/DSegebarth/dcl_stats_n_plots/master/test_data/independent_samples_3groups.csv&#39;</span>



<span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">file_url</span><span class="p">)</span>
<span class="n">df</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>

    </div>
</div>
</div>

</div>
    {% endraw %}

<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>To perform the statistical analysis, all we now have to do is pass this data as Pandas DataFrame to the <a href="/dcl_stats_n_plots/stats.html#independent_samples"><code>independent_samples</code></a> function of the stats module.
This will return a dictionary containing the results of the statistical analysis, but also some automatically extracted parameters that will be of use for both plotting and annotation of the data later on. For the exact details of which statistical tests are performed, please head over to the documentation of the <a href="/dcl_stats_n_plots/stats.html#independent_samples"><code>independent_samples</code></a> function. In brief:</p>
<ol>
<li>The prerequisites for parametric testing (normal distribution and equality of variances) are assessed</li>
<li>Parametric or non-parametric tests are used accordingly</li>
<li>If the dataset contains more than two groups, pairwise comparisons of all possible combinations are performed and the results are corrected for multiple comparisons</li>
</ol>
<p>The results of the pairwise comparisons can be accessed as follows:</p>

</div>
</div>
</div>
    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="kn">from</span> <span class="nn">dcl_stats_n_plots</span> <span class="kn">import</span> <span class="n">stats</span>

<span class="n">results</span> <span class="o">=</span> <span class="n">stats</span><span class="o">.</span><span class="n">independent_samples</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>

<span class="n">results</span><span class="p">[</span><span class="s1">&#39;summary&#39;</span><span class="p">][</span><span class="s1">&#39;pairwise_comparisons&#39;</span><span class="p">]</span>
</pre></div>

    </div>
</div>
</div>

</div>
    {% endraw %}

<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>If there were more than two groups present, the results of the group level statistical test can be accessed as:</p>

</div>
</div>
</div>
    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">results</span><span class="p">[</span><span class="s1">&#39;summary&#39;</span><span class="p">][</span><span class="s1">&#39;group_level_statistic&#39;</span><span class="p">]</span>
</pre></div>

    </div>
</div>
</div>

</div>
    {% endraw %}

<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>{% include note.html content='Again, for more detailed information about the statistical tests, as well as the generated output, please have a look at the documentation of the <a href="/dcl_stats_n_plots/stats.html#independent_samples"><code>independent_samples</code></a> function. Here you will also find links to the corresponding functions of the python package "pingouin", which is used for statistical testing.' %}</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Plot-the-data-and-annotate-the-stats:">Plot the data and annotate the stats:<a class="anchor-link" href="#Plot-the-data-and-annotate-the-stats:"> </a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>With the <code>results</code> dictionary and our dataset as Pandas DataFrame, we have everything we need to plot the data and annotate the stats in one go.</p>

</div>
</div>
</div>
    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="kn">from</span> <span class="nn">dcl_stats_n_plots</span> <span class="kn">import</span> <span class="n">plots</span>

<span class="n">plots</span><span class="o">.</span><span class="n">plot_independent_samples</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">plot_type</span> <span class="o">=</span> <span class="s1">&#39;stripplot&#39;</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
    {% endraw %}

<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>For all types of statistical analyses that are implemented in dcl_stats_n_plots, we also provide several options of how to visualize your data. In order to change the type of plot, simply choose from the range of currently supported plots (see <a href="/dcl_stats_n_plots/plots.html#plot_independent_samples"><code>plot_independent_samples</code></a>) and change the string you pass to the function accordingly:</p>

</div>
</div>
</div>
    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">plots</span><span class="o">.</span><span class="n">plot_independent_samples</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">plot_type</span> <span class="o">=</span> <span class="s1">&#39;violinplot with stripplot overlay&#39;</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
    {% endraw %}

<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>{% include warning.html content='The annotation of the statistical results via the API is not yet implemented. This is one of the next features that will be added within one of the following patch releases - if not <em>the</em> next.' %}</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Customize-the-plot">Customize the plot<a class="anchor-link" href="#Customize-the-plot"> </a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>dcl_stats_n_plots provides you with a bunch of customization options. A comprehensive list will be made available <a href="https://dsegebarth.github.io/dcl_stats_n_plots/plots.html">here</a> shortly. Probably the easiest way to explore them all is by using our GUI, where you can play arround with all parameters and interactively change them back and forth until you found the perfect settings for your plot. Since the customization parameters work in the same manner for all types of analyses and plots, you can also have a look at the other provided customization examples.</p>
<p>In order to customize your plots using our API, you have to pass a dictionary to the plotting function, which contains all parameters that you would like to change. We designed our API in such a way, that you only have to pass those parameters to the function, which you want to customize. The default settings will be used for all remaining parameters.</p>
<p>For instance, you can specify the exact label of the x- and y-axis, change their fontsize, increase the marker size of the stripplot overlay, use a different color palette for the violin plots, and finally adjust the figure size, as per individual demand.</p>

</div>
</div>
</div>
    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">params</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;yaxis_label_text&#39;</span><span class="p">:</span> <span class="s1">&#39;time spent in center [s]&#39;</span><span class="p">,</span>
          <span class="s1">&#39;yaxis_label_fontsize&#39;</span><span class="p">:</span> <span class="mi">15</span><span class="p">,</span>
          <span class="s1">&#39;xaxis_label_text&#39;</span><span class="p">:</span> <span class="s1">&#39;genotype&#39;</span><span class="p">,</span>
          <span class="s1">&#39;xaxis_label_fontsize&#39;</span><span class="p">:</span> <span class="mi">15</span><span class="p">,</span>
          <span class="s1">&#39;marker_size&#39;</span><span class="p">:</span> <span class="mi">8</span><span class="p">,</span>
          <span class="s1">&#39;color_palette&#39;</span><span class="p">:</span> <span class="s1">&#39;rocket&#39;</span><span class="p">,</span>
          <span class="s1">&#39;fig_width&#39;</span><span class="p">:</span> <span class="mi">20</span><span class="p">,</span>
          <span class="s1">&#39;fig_height&#39;</span><span class="p">:</span> <span class="mi">15</span><span class="p">}</span>
          
<span class="c1"># Pass the dictionary as &#39;params&#39; when calling the plotting function</span>
<span class="n">plots</span><span class="o">.</span><span class="n">plot_independent_samples</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">plot_type</span> <span class="o">=</span> <span class="s1">&#39;violinplot with stripplot overlay&#39;</span><span class="p">,</span> <span class="n">params</span> <span class="o">=</span> <span class="n">params</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
    {% endraw %}

<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Comparison-of-a-single-group-to-a-fixed-value">Comparison of a single group to a fixed value<a class="anchor-link" href="#Comparison-of-a-single-group-to-a-fixed-value"> </a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Representative-experiment">Representative experiment<a class="anchor-link" href="#Representative-experiment"> </a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>{% include note.html content='will be added soon.' %}</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Expected-input-data-format">Expected input data format<a class="anchor-link" href="#Expected-input-data-format"> </a></h3><p>For this analysis, you have to provide three columns. The first column must contain the data, the second column provides the corresponding group_id (see note below the image), and the third column provides the fixed value against which the group shall be compared.</p>
<p>In Microsoft Excel, this might look like the following:</p>
<p><img src="https://raw.githubusercontent.com/DSegebarth/dcl_stats_n_plots/master/media/one_sample_input_schema.png" alt="one_sample_input_schema.png"></p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>{% include note.html content='Of course, the second column (&#8217;group column&#8217;) will only contain one group_id, since this is a <em>one sample</em> analysis. However, keeping the expected input data structure consistent across functions (data, group, ..) is supposed to make the use of dcl_stats_n_plots a little easier for the user. On top, it allows the user to provide the x-axis-label (= group column header) and the corresponding x-axis-tick-label (= the group_id that is provided in the group column).' %}</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Compute-the-stats">Compute the stats<a class="anchor-link" href="#Compute-the-stats"> </a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>In order to perform the statistical analysis, simply pass the data in a correctly structured pandas DataFrame to our <a href="/dcl_stats_n_plots/stats.html#one_sample"><code>one_sample</code></a> function of the stats module. Please also check out the documentation there for more detailed information about what statistical tests are run.</p>

</div>
</div>
</div>
    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>

<span class="n">file_url</span> <span class="o">=</span> <span class="s1">&#39;https://raw.githubusercontent.com/DSegebarth/dcl_stats_n_plots/master/test_data/one_sample_significant.csv&#39;</span>

<span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">file_url</span><span class="p">)</span>
<span class="n">df</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>

    </div>
</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="kn">from</span> <span class="nn">dcl_stats_n_plots</span> <span class="kn">import</span> <span class="n">stats</span>

<span class="n">results</span> <span class="o">=</span> <span class="n">stats</span><span class="o">.</span><span class="n">one_sample</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>

<span class="n">results</span><span class="p">[</span><span class="s1">&#39;summary&#39;</span><span class="p">][</span><span class="s1">&#39;pairwise_comparisons&#39;</span><span class="p">]</span>
</pre></div>

    </div>
</div>
</div>

</div>
    {% endraw %}

<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Plot-the-data-and-annotate-the-stats:">Plot the data and annotate the stats:<a class="anchor-link" href="#Plot-the-data-and-annotate-the-stats:"> </a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>Using the <code>results</code> dictionary, we now have the option plot our data and annotate the results of the statistical analysis in one go.</p>

</div>
</div>
</div>
    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="kn">from</span> <span class="nn">dcl_stats_n_plots</span> <span class="kn">import</span> <span class="n">plots</span>

<span class="n">plots</span><span class="o">.</span><span class="n">plot_one_sample</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">plot_type</span> <span class="o">=</span> <span class="s1">&#39;stripplot&#39;</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
    {% endraw %}

<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>{% include warning.html content='The annotation of the statistical results via the API is not yet implemented. This is one of the next features that will be added within one of the following patch releases - if not <em>the</em> next.' %}</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Customize-the-plot">Customize the plot<a class="anchor-link" href="#Customize-the-plot"> </a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>You also have the chance to customize several features of the generated plot, simply by specifying the respective features in a dictionary and passing it along when you call the plotting function.</p>

</div>
</div>
</div>
    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">params</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;color_palette&#39;</span><span class="p">:</span> <span class="p">{</span><span class="s1">&#39;group_a&#39;</span><span class="p">:</span> <span class="s1">&#39;forestgreen&#39;</span><span class="p">},</span>
          <span class="s1">&#39;marker_size&#39;</span><span class="p">:</span> <span class="mi">7</span><span class="p">,</span>
          <span class="s1">&#39;fig_width&#39;</span><span class="p">:</span> <span class="mi">14</span><span class="p">,</span>
          <span class="s1">&#39;yaxis_scaling_mode&#39;</span><span class="p">:</span> <span class="s1">&#39;manual&#39;</span><span class="p">,</span>
          <span class="s1">&#39;yaxis_lower_lim_value&#39;</span><span class="p">:</span> <span class="mi">0</span><span class="p">,</span>
          <span class="s1">&#39;yaxis_upper_lim_value&#39;</span><span class="p">:</span> <span class="mi">100</span><span class="p">,</span>
          <span class="s1">&#39;yaxis_label_text&#39;</span><span class="p">:</span> <span class="s1">&#39;time spent in target half [%]&#39;</span><span class="p">,</span>
          <span class="s1">&#39;xaxis_label_text&#39;</span><span class="p">:</span> <span class="s1">&#39;&#39;</span><span class="p">}</span>

<span class="n">plots</span><span class="o">.</span><span class="n">plot_one_sample</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">plot_type</span> <span class="o">=</span> <span class="s1">&#39;boxplot with stripplot overlay&#39;</span><span class="p">,</span> <span class="n">params</span> <span class="o">=</span> <span class="n">params</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
    {% endraw %}

<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Mixed-Model-ANOVA">Mixed-Model ANOVA<a class="anchor-link" href="#Mixed-Model-ANOVA"> </a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>{% include warning.html content='This function is currently only implemented in a parametric variant (there is no non-parametric equivalent available via pingouin). This affects only the group level statistics (main effects), while pairwise comparisons are again performed either using parametric or non-parametric tests, depending on the respective data.' %}</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Representative-experiment">Representative experiment<a class="anchor-link" href="#Representative-experiment"> </a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>{% include note.html content='will be added soon.' %}</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Expected-input-data-format">Expected input data format<a class="anchor-link" href="#Expected-input-data-format"> </a></h3><p>Now, for this analysis, things get a little bit more complicated. In total, your input data has to be structured in four columns (also have a look at the image below). First and second column are again, as for the other analyses, the columns that provide the data values (1st) and group assignment (2nd). The third column has to provide unique subject identifiers for each experimental subject. The values of the fourth column reflect the corresponding recording sessions (i.e. an identifier for each part of your experiment, in which one of the repeated measurements took place).</p>
<p>In Microsoft Excel, this might look like the following:</p>
<p><img src="https://raw.githubusercontent.com/DSegebarth/dcl_stats_n_plots/master/media/mma_input_schema.png" alt="mma_input_schema.png"></p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Compute-the-stats">Compute the stats<a class="anchor-link" href="#Compute-the-stats"> </a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>In order to perform the statistical analysis, simply pass the data in a correctly structured pandas DataFrame to the <a href="/dcl_stats_n_plots/stats.html#mixed_model_ANOVA"><code>mixed_model_ANOVA</code></a> function of the stats module. Please also check out the documentation there for more detailed information about what statistical tests are run.</p>

</div>
</div>
</div>
    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>

<span class="n">file_url</span> <span class="o">=</span> <span class="s1">&#39;https://raw.githubusercontent.com/DSegebarth/dcl_stats_n_plots/master/test_data/mixed_model_anova_3groups.csv&#39;</span>

<span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">file_url</span><span class="p">)</span>
<span class="n">df</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>

    </div>
</div>
</div>

</div>
    {% endraw %}

<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>Since this function is currently only implemented in a parametric variant, calling the function will prompt you with a warning about this if your data actually violate the pre-requisites for parametric testing.</p>

</div>
</div>
</div>
    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="kn">from</span> <span class="nn">dcl_stats_n_plots</span> <span class="kn">import</span> <span class="n">stats</span>

<span class="n">results</span> <span class="o">=</span> <span class="n">stats</span><span class="o">.</span><span class="n">mixed_model_ANOVA</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
    {% endraw %}

<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>To access the group level statistics of the Mixed-Model ANOVA:</p>

</div>
</div>
</div>
    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">results</span><span class="p">[</span><span class="s1">&#39;summary&#39;</span><span class="p">][</span><span class="s1">&#39;group_level_statistic&#39;</span><span class="p">]</span>
</pre></div>

    </div>
</div>
</div>

</div>
    {% endraw %}

<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>And to inspect the pairwise comparisons in detail:</p>

</div>
</div>
</div>
    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">results</span><span class="p">[</span><span class="s1">&#39;summary&#39;</span><span class="p">][</span><span class="s1">&#39;pairwise_comparisons&#39;</span><span class="p">]</span>
</pre></div>

    </div>
</div>
</div>

</div>
    {% endraw %}

<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>{% include note.html content='Please note, that even though the data violated the pre-requisites for parametric testing on the group-level, they were again met for all pairwise comparisons and, thus, parametric testing was used in all these cases. ' %}</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Plot-the-data-and-annotate-the-stats:">Plot the data and annotate the stats:<a class="anchor-link" href="#Plot-the-data-and-annotate-the-stats:"> </a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>Using the <code>results</code> dictionary, we now have the option plot our data and annotate the results of the statistical analysis in one go. The 'pointplot' option is good to visualize the progression over the course of the recorded sessions:</p>

</div>
</div>
</div>
    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="kn">from</span> <span class="nn">dcl_stats_n_plots</span> <span class="kn">import</span> <span class="n">plots</span>

<span class="n">plots</span><span class="o">.</span><span class="n">plot_mma</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">plot_type</span> <span class="o">=</span> <span class="s1">&#39;pointplot&#39;</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
    {% endraw %}

<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>However, sometimes there may not be an immediate connection between the individual sessions (which is why the connecting lines should be avoided), or you simply want to show all individual data points, instead of the group mean with an error bar, like for the pointplot. For this, feel free to check out the other plotting options we implemented to visualize your Mixed-Model ANOVA data:</p>

</div>
</div>
</div>
    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">plots</span><span class="o">.</span><span class="n">plot_mma</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">plot_type</span> <span class="o">=</span> <span class="s1">&#39;boxplot with stripplot overlay&#39;</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
    {% endraw %}

<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>{% include warning.html content='The annotation of the statistical results via the API is not yet implemented. This is one of the next features that will be added within one of the following patch releases - if not <em>the</em> next.' %}</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Customize-the-plot">Customize the plot<a class="anchor-link" href="#Customize-the-plot"> </a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>You also have the chance to customize several features of the generated plot, simply by specifying the respective features in a dictionary and passing it along when you call the plotting function. For instance, you can also re-arrange the order of the x-axis labels or of the groups.</p>

</div>
</div>
</div>
    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">params</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;l_xlabel_order&#39;</span><span class="p">:</span> <span class="p">[</span><span class="s1">&#39;session_3&#39;</span><span class="p">,</span> <span class="s1">&#39;session_2&#39;</span><span class="p">,</span> <span class="s1">&#39;session_1&#39;</span><span class="p">,</span> <span class="s1">&#39;session_0&#39;</span><span class="p">],</span>
          <span class="s1">&#39;l_hue_order&#39;</span><span class="p">:</span> <span class="p">[</span><span class="s1">&#39;group_b&#39;</span><span class="p">,</span> <span class="s1">&#39;group_c&#39;</span><span class="p">,</span> <span class="s1">&#39;group_a&#39;</span><span class="p">],</span>
          <span class="s1">&#39;color_palette&#39;</span><span class="p">:</span> <span class="s1">&#39;viridis&#39;</span><span class="p">}</span>

<span class="n">plots</span><span class="o">.</span><span class="n">plot_mma</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">plot_type</span> <span class="o">=</span> <span class="s1">&#39;violinplot with stripplot overlay&#39;</span><span class="p">,</span> <span class="n">params</span> <span class="o">=</span> <span class="n">params</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
    {% endraw %}

<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>{% include warning.html content='To be continued...' %}</p>

</div>
</div>
</div>
</div>
 

